### ISSUE TITLE
[Feature Request] Transition timing according to sunset/sunrise times

### ISSUE THREAD

**Issue by jpbaril** [jacob-pro/solar-screen-brightness#34]
Hi,
First, thank you for this great piece of software.

One thing I noticed is that sunset time is always at 50 % of transition duration.
So if I set a 120 minutes transition, then 1 hour before sunset time the screen brightness will begin to be decreased and will continue to decreased until 1 hour after sunset time.

I would like to be able to set that sunset time be at, for example, 10 % of transition duration. That is that screen brightness begins to decrease only just a few minutes before sunset time.

**Comment by caesay** [comment]
I agree, I don't need anything as granular as #35, but being able to add a start and end offset would be tremendously useful. By the time the sunset "time" rolls around, the sun has been completely down for me for some time, so I already want to be at my night brightness beforehand. 

Similarly for sunrise, the sun is still not visible at the "sunrise time" so I'd like to start increasing brightness _after_ that time. 

This would be possible with some simple offsets - eg. 
- Transition time 60min
- Sunset offset -30min
- Sunset offset +30min
